### ISSUE TITLE
After updating to 4.3.0, attempting to get the network response when a filter is applied results in an error "Cannot read property trim of undefined"

### ISSUE THREAD

**Issue by chadbellangivelify** [webdriverio-community/wdio-intercept-service#416]
The call I'm making is:

let capturedNetworkCall = browser.getRequests().filter(request => request.method === requestType && request.url.includes(endpointIncludes));

It is working with wdio-intercept-service 4.2.2 and failing with version 4.3.0

The error is:
[chrome 106.0.5249.61 mac os x #0-10] Cannot read property 'trim' of undefined
[chrome 106.0.5249.61 mac os x #0-10] TypeError: Cannot read property 'trim' of undefined
[chrome 106.0.5249.61 mac os x #0-10]     at parseResponseHeaders (/srv/analytics-studio-automated-tester/releases/20221017235711/node_modules/wdio-intercept-service/index.js:351:32)
[chrome 106.0.5249.61 mac os x #0-10]     at transformRequest (/srv/analytics-studio-automated-tester/releases/20221017235711/node_modules/wdio-intercept-service/index.js:327:20)
[chrome 106.0.5249.61 mac os x #0-10]     at Array.map (<anonymous>)
[chrome 106.0.5249.61 mac os x #0-10]     at getRequest (/srv/analytics-studio-automated-tester/releases/20221017235711/node_modules/wdio-intercept-service/index.js:295:24)
[chrome 106.0.5249.61 mac os x #0-10]     at runMicrotasks (<anonymous>)
[chrome 106.0.5249.61 mac os x #0-10]     at Function.getNetworkResponse (/srv/analytics-studio-automated-tester/releases/20221017235711/helpers/utilities.js:407:43)

**Comment by tehhowch** [comment]
That is this line: https://github.com/webdriverio-community/wdio-intercept-service/blob/057537d0f99b12274d553651da434665f3f8f8e9/index.js#L351

mind sharing the headers on the request that is causing the issue?
cc @jbebe

**Comment by jbebe** [comment]
If the header contains a colon, undefined will not occur, even if there is no content after the colon. Given the information there can only be one problem: one of the http headers does not contain a colon. Which is very weird so ~~we~~ you need more context here.

**Comment by sumanthpfs** [comment]
I get the same error too, after updating to 4.3. 
I was unable to do any assertions or get methods on the requests.

**Comment by tehhowch** [comment]
@sumanthpfs @chadbellangivelify please provide the headers that are causing the issue. You can determine which header is responsible by making a modification in the library's downloaded package (inside `node_modules`) as [identified above](https://github.com/webdriverio-community/wdio-intercept-service/issues/416#issuecomment-1289631918)

**Comment by chadbellangivelify** [comment]
At this time, I'm no longer able to reproduce this issue.  Maybe Sumanthpfs can provide more details to help

**Comment by dragosMC91** [comment]
for me it's a request like this causing the problem (added a console log):
```
{
    __fulfilled: 1669122851503,
    __processed: 1669122851372,
    body: '',
    headers: {},
    method: 'POST',
    requestBody: '[{"event":"map.load","created":"2022-11-22T13:14:11.371Z"}]',
    requestHeaders: { 'content-type': 'text/plain' },
    statusCode: 204,
    url: 'https://events.mapbox.com/events/v2'
}
```
removed some sensitive info from the url and body.

For the mentioned request, I get the error when `parseResponseHeaders` is called.

I could investigate next weekend if nobody fixes it by then

**Comment by dragosMC91** [comment]
im providing a reproductible example here: 
1. checkout this branch: https://github.com/dragosMC91/wdio-exclude-pattern-bug-highlight/pull/3
2. (using node 16.18.0) `yarn install`
3. run `yarn test --spec test/specs/suite1/interceptorError.mobile.test.js`

Running this test triggers the mentioned error.

**Comment by vlohana** [comment]
Any solution for this issue? I am getting it very often for my application!

**Comment by tehhowch** [comment]
@vlohana please upgrade to v4.3.1, released earlier today

**Comment by tehhowch** [comment]
closed by #438
